app/core/datastore/src/main/kotlin/com/hedvig/android/core/datastore/SettingsDataStore.kt

```diff
@@ -0,0 +1,44 @@
+package com.hedvig.android.core.datastore
+
+import androidx.datastore.core.DataStore
+import androidx.datastore.preferences.core.Preferences
+import androidx.datastore.preferences.core.edit
+import androidx.datastore.preferences.core.stringPreferencesKey
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.map
+
+interface SettingsDataStore {
+  suspend fun setTheme(theme: String)
+  fun observeTheme(): Flow<String>
+  suspend fun setLanguage(language: String)
+  fun observeLanguage(): Flow<String>
+}
+
+internal class SettingsDataStoreImpl(
+  private val dataStore: DataStore<Preferences>,
+) : SettingsDataStore {
+
+  private val languageKey = stringPreferencesKey("settings-language")
+  private val themeKey = stringPreferencesKey("settings-theme")
+
+  override suspend fun setTheme(theme: String) {
+    dataStore.edit {
+      it[themeKey] = theme
+    }
+  }
+
+  override fun observeTheme(): Flow<String> {
```

Comment:
Do you think this could return an instance of our `enum class Theme` directly? So that we can use the default of Theme.Light for example instead of the String "LIGHT", and also not have to put the burden of doing this translation on the consumer of this class.

This might require that Theme enum to move to another module and that's perfectly fine. It could move into a app:theme, exactly like we have app:market and app:language.
Then this datastore module could depend on that so it knows about Theme, and :app:app would also be able to know about it and use it.

But could be another PR and if you feel like doing it.

Reply:
Should definitely do that, ill add another commit